From bedc79ee6d396e60800a6ae8c47bc2e153cfa726 Mon Sep 17 00:00:00 2001
From: David Redondo <kde@david-redondo.de>
Date: Wed, 24 Sep 2025 17:23:16 +0200
Subject: [PATCH] kcms/mouse: Make button rebinding work again

2ec4cdfe92bcba736fa64b4fc6e534cfc15abb59 in Kirigami broke it
for some reason, event handling code looks the same.
For now as a workaround we can make the columnview non-interactive
while listening for mouse buttons.
BUG:506342
FIXED-IN:6.4.6


(cherry picked from commit 84f1f2f3cd62373b76d759f6a3e95f25b74b97cb)

Co-authored-by: David Redondo <kde@david-redondo.de>

(cherry picked from commit 756a579f4a925158a485aeff45144b02d5e2145c)

Co-authored-by: David Redondo <kde@david-redondo.de>
---
 kcms/mouse/ui/bindings.qml | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/kcms/mouse/ui/bindings.qml b/kcms/mouse/ui/bindings.qml
index 4d66fbda80..0abb179d85 100644
--- a/kcms/mouse/ui/bindings.qml
+++ b/kcms/mouse/ui/bindings.qml
@@ -31,6 +31,13 @@ KCMUtils.SimpleKCM {
         hotplugMessage: root.KCMUtils.ConfigModule.hotplugMessage
     }
 
+    Kirigami.ColumnView.onViewChanged: {
+        const view = Kirigami.ColumnView.view
+        if (view) {
+            view.interactive = Qt.binding(() => !newBinding.checked)
+        }
+    }
+
     MouseArea {
         // Deliberately using MouseArea on the page instead of a TapHandler on the button, so we can capture clicks anywhere
         id: buttonCapture
-- 
2.51.0

